src/planning/path_planner/src/path_planner.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Package:   path_planner
+ * Filename:  path_planner.cpp
+ * Author:    Jim Moore
+ * Email:     jim3moore@gmail.com
+ * Copyright: 2021, Nova UTD
+ * License:   MIT License
+ */
+
+#include "path_planner/path_planner.hpp"
+#include "path_planner/map_utils.hpp"
+
+#include "autoware_auto_msgs/msg/trajectory.hpp"
+#include <lanelet2_core/geometry/Lanelet.h>
+#include <lanelet2_core/LaneletMap.h>
+#include <lanelet2_core/primitives/Point.h>
+#include <lanelet2_core/geometry/LineString.h>
+#include <autoware_auto_msgs/msg/trajectory.hpp>
+#include <autoware_auto_msgs/msg/had_map_route.hpp>
+#include <had_map_utils/had_map_utils.hpp>
+#include <common/types.hpp>
+#include <geometry/common_2d.hpp>
+
+using namespace autoware::common::types;
+using TrajectoryPoint = autoware_auto_msgs::msg::TrajectoryPoint;
+using autoware_auto_msgs::msg::HADMapRoute;
+using autoware_auto_msgs::msg::Trajectory;
+
+using namespace navigator::path_planner;
+
+
+
+size_t get_closest_lanelet(const lanelet::ConstLanelets &lanelets, const autoware_auto_msgs::msg::TrajectoryPoint &point)
+{
+    float64_t closest_distance = std::numeric_limits<float64_t>::max();
+    size_t closest_index = 0;
+    for (size_t i = 0; i < lanelets.size(); i++)
+    {
+        const auto &llt = lanelets.at(i);
+        const auto &point2d = lanelet::Point2d(lanelet::InvalId, point.x, point.y).basicPoint2d();
+        // TODO(mitsudome-r): change this implementation to remove dependency to boost
+        const float64_t distance = lanelet::geometry::distanceToCenterline2d(llt, point2d);
+        if (distance < closest_distance)
+        {
+            closest_distance = distance;
+            closest_index = i;
+        }
+    }
+    return closest_index;
+}
+lanelet::Point3d convertToLaneletPoint(
+    const autoware_auto_msgs::msg::TrajectoryPoint &pt)
+{
+    return lanelet::Point3d(lanelet::InvalId, pt.x, pt.y, 0.0);
+}
+ParameterizedSpline PathPlanner::get_center_line_spline(const std::vector<autoware_auto_msgs::msg::TrajectoryPoint> &line_points)
+{
+    std::vector<double> x_points(line_points.size());
+    std::vector<double> y_points(line_points.size());
+    for (const auto &p : line_points)
+    {
+        x_points.push_back(p.x);
+        y_points.push_back(p.y);
+    }
+    return ParameterizedSpline(x_points, y_points);
+}
+autoware_auto_msgs::msg::TrajectoryPoint convertToTrajectoryPoint(const lanelet::ConstPoint3d &pt)
+{
+    autoware_auto_msgs::msg::TrajectoryPoint trajectory_point;
+    trajectory_point.x = static_cast<float32_t>(pt.x());
+    trajectory_point.y = static_cast<float32_t>(pt.y());
+    trajectory_point.longitudinal_velocity_mps = 0;
+    return trajectory_point;
+}
+//resolution should be < 0.1 to get good approximation
```

Comment:
0.1 what? Meters? You used "units" earlier, but the Lanelet data is stored in meters, right?

Reply:
It should be 0.1 of whatever units we decide to use, so meters is fine. It's more of a note to myself, for the future from my testing earlier. Will be more robust later.